crates/recursor/src/lib.rs

```diff
@@ -85,6 +90,77 @@ impl DnssecPolicy {
     }
 }
 
+/// An IPv4/IPv6 access control set.  This mainly hides the complexity of supporting v4 and v6
+/// addresses concurrently in a given set.  The AccessControlSet differs from a typical Access
+/// Control List in that there is no order.  The access semantics are:
+///
+/// | Present in allow list | Present in deny list |  Result  |
+/// |-----------------------|----------------------|----------|
+/// |                  true |                false |  allowed |
+/// |                  true |                 true |  allowed |
+/// |                 false |                false |  allowed |
```

Comment:
given the semantics when not found in the allow list, maybe it's better described as an "deny override list"? Mostly just spit-balling, I'm not sure it's a clear improvement.